### PR TITLE
Change How the Custom pre-commit Configuration for GitHub Actions is Handled and Fix terraform-docs Installation

### PR DESCRIPTION
**Pull request by mcdonnnj**
# 

## 🗣 Description
After reflection about future technical debt [in another PR](https://github.com/cisagov/skeleton-packer/pull/34#discussion_r421030954), we changed how a custom GitHub Actions `pre-commit` configuration would be handled. This PR implements the change approved in the [aforementioned cisagov/skeleton-packer PR](https://github.com/cisagov/skeleton-packer/pull/34).

This PR also fixes an issue observed with recent workflows that install `terraform-docs`. @dav3r observed it failing in GitHub Actions and tracked down the issue. The changes he implemented are implemented here in turn.

## 💭 Motivation and Context
Maintaining two `pre-commit` configurations could be troublesome in the future. We will now create a temporary `pre-commit` configuration file with the offending hook (`terraform_validate`) removed in the workflow. This temporary configuration is then used by the `pre-commit run --all-files` call in the workflow to get the desired functionality.

GitHub Actions workflows using `terraform-docs` no longer worked, we had to fix it 🔨

## 🧪 Testing
Automated testing passes.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -91,17 +91,23 @@ jobs:
...
+          > /tmp/.github-actions-pre-commit-config.yaml
+      - name: Run pre-commit with modified configuration file on all files
         # We run pre-commit here with a custom configuration that has
         # the terraform-validate hook removed.  This is because
         # terraform validate cannot currently run without accessing
         # the remote state because of the way the providers are
-        # defined in terraform/providers.tf.  This is something that
-        # may be remedied in the future.  For more information, check
-        # out these two GitHub issues:
+        # defined in providers.tf.  This is something that may be
```
Comment (dav3r) [inline .github/workflows/build.yml]: This repo actually has a `terraform/providers.tf` file, so it's ok to leave that part in.

**Review by dav3r — APPROVED** [review]
👍 but please correct the small item I mentioned in my comment.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml]: Oops, I was a little too blanket in my copy/paste on these PRs. Fixed in dac1116.

**Review by jsf9k — APPROVED** [review]
My god!  It's full of stars!

**Pull request merged**